### PR TITLE
QA-14836 : add cypress test

### PR DESCRIPTION
**Pull request by ksagnard**
## Summary by CodeRabbit

- **Tests**
	- Added a new test case to verify user permissions for viewing and editing data in the vanity URL modal.
	- Introduced a function to check the visibility of the vanity URL menu item in the content editor.

### CODE REVIEW

File: tests/cypress/e2e/permissions/checkVanityUrlModalPermissions.cy.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        deleteNode('/roles/noVanityRole')
+    })
+
+    it('Verify that user having permission can see and edit some data in the vanity url modal', function () {
+        cy.login('user1', 'password')
+        const jcontent = JContent.visit(siteKey, 'en', 'pages/home')
+        jcontent.switchToListMode()
+        jcontent.editComponentByText(pageNameA)
+        const contenteditor = new ContentEditorSEO()
+        contenteditor.checkVanityUrlAccess(false)
```
Comment (jsinovassin) [inline tests/cypress/e2e/permissions/checkVanityUrlModalPermissions.cy.ts:139]: It seems that you are checking that the button is not visible here, but you click on it at next line

File: tests/cypress/page-object/ContentEditorSEO.ts [hunk 2]
```diff
@@ -16,4 +16,13 @@ export class ContentEditorSEO extends ContentEditor {
...
         cy.get('li[data-sel-role="vanityUrls"]').should('have.attr', 'aria-disabled', enabled)
     }
+
+    checkVanityUrlAccess(visible) {
+        cy.get('button[data-sel-role="3dotsMenuAction"]').click()
+        if ('true' == visible) {
+            cy.get('li[data-sel-role="vanityUrls"]').should('be.visible')
+        } else {
+            cy.get('li[data-sel-role="vanityUrls"]').should('not.be.visible')
+        }
```
Comment (jsinovassin) [inline tests/cypress/page-object/ContentEditorSEO.ts:26]: it's possible to simplify these lines by
`cy.get('li[data-sel-role="vanityUrls"]').should(visible ? 'be.visible' : 'not.be.visible')`

File: tests/cypress/page-object/ContentEditorSEO.ts [hunk 3]
```diff
@@ -16,4 +16,13 @@ export class ContentEditorSEO extends ContentEditor {
         cy.get('button[data-sel-role="3dotsMenuAction"]').click()
         cy.get('li[data-sel-role="vanityUrls"]').should('have.attr', 'aria-disabled', enabled)
     }
+
+    checkVanityUrlAccess(visible) {
```
Comment (jsinovassin) [inline tests/cypress/page-object/ContentEditorSEO.ts:20]: Can we find a better name for this function there is already checkVanityUrlAccessibility in this file.
A name like shouldVanityUrlMenuVisible or better

File: tests/cypress/e2e/permissions/checkVanityUrlModalPermissions.cy.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    deleteUser,
+    grantRoles,
+    addNode,
+    deleteNode,
+} from '@jahia/cypress'
+import { JContent } from '@jahia/jcontent-cypress/dist/page-object/jcontent'
+import { ContentEditorSEO } from '../../page-object/ContentEditorSEO'
+import { checkVanityUrlByAPI } from '../../utils/Utils'
+
+describe('Test UIs permissions', () => {
```
Comment (jsinovassin) [inline tests/cypress/e2e/permissions/checkVanityUrlModalPermissions.cy.ts:15]: There is also the file checkUisPermissions.cy.ts where there is some tests on the vanity url modale. Did you ensure that they are still valid and that there is no duplication between both?

**Review by jsinovassin — CHANGES_REQUESTED** [review]

File: tests/cypress/e2e/permissions/checkVanityUrlModalPermissions.cy.ts [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+                {
+                    name: 'j:permissionNames',
+                    values: ['jcr:all_default', 'viewVanityUrlModal'],
+                },
+                { name: 'j:hidden', value: 'false' },
+                { name: 'j:roleGroup', value: 'edit-role' },
+                { name: 'j:privilegedAccess', value: 'true' },
+            ],
+        })
+        addNode({
```
[the remark this answers is not in the document]
Comment (ksagnard) [reply tests/cypress/e2e/permissions/checkVanityUrlModalPermissions.cy.ts]: done

File: tests/cypress/e2e/permissions/checkVanityUrlModalPermissions.cy.ts [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+        const jcontent = JContent.visit(siteKey, 'en', 'pages/home')
+        jcontent.switchToListMode()
+        jcontent.editComponentByText(pageNameA)
+        const contenteditor = new ContentEditorSEO()
+        contenteditor.checkVanityUrlAccess(false)
+        const vanityUrlUi = contenteditor.openVanityUrlUi()
+        vanityUrlUi.addVanityUrl('vanityurl-test-modalAccess')
+
+        // Check vanity url was created as expected
+        checkVanityUrlByAPI(
```
[the remark this answers is not in the document]
Comment (ksagnard) [reply tests/cypress/e2e/permissions/checkVanityUrlModalPermissions.cy.ts:147]: done

File: tests/cypress/e2e/permissions/checkVanityUrlModalPermissions.cy.ts (hunk 4 above)
Comment (ksagnard) [reply tests/cypress/e2e/permissions/checkVanityUrlModalPermissions.cy.ts:15]: yes : they are not testing the same right : siteAdminUrlmapping

File: tests/cypress/page-object/ContentEditorSEO.ts (hunk 2 above)
Comment (ksagnard) [reply tests/cypress/page-object/ContentEditorSEO.ts]: done

**Review by jsinovassin — APPROVED** [review]

**Pull request merged**